.pre-commit-config.yaml

```diff
@@ -0,0 +1,37 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+
+  - repo: local
+    hooks:
+      - id: clang-format-fix
+        name: clang-format-fix
+        entry: clang-format-18
+        files: ^(src/|include/)
+        types_or: [c++, c]
+        language: system
+        args: ['-i']
+        stages: [manual]
+
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.3.5
+    hooks:
+      - id: clang-format
+        additional_dependencies: ['clang-format==18.1.8']
+      - id: clang-tidy
+        stages: [manual]
+        additional_dependencies: ['clang-tidy==18.1.8']
+
+  - repo: https://github.com/cheshirekow/cmake-format-precommit
+    rev: "v0.6.13"
+    hooks:
+      - id: cmake-format
+        additional_dependencies: [pyyaml]
+      - id: cmake-lint
```

Comment:
Latest release in 4 years. We might want to find an alternative.

Reply:
This project is a configuration placement of [`cmake-format`](https://github.com/cheshirekow/cmake_format). 
Which seem to be abandoned, see: https://github.com/cheshirekow/cmake_format/issues/340

Reply:
omitted, let's find something better. Issue raised to optional project at: https://github.com/beman-project/optional26/issues/67